### DISCUSSION TITLE
Export .mesh to a VisIt format?

### DISCUSSION THREAD

**Discussion by wnissen** [mfem/mfem#3397] [Q&A]
What's the best way to export an MFEM .mesh formatted file to VisIt or something else that will let me run some other metrics on it beyond what is in `mesh-explorer`?

**Comment by jandrej** [comment]
You can use `DataCollection`. See https://github.com/mfem/mfem/blob/master/examples/ex9.cpp#L331-L351

**Reply by wnissen** [reply]
Thanks, can I stick that bit of code in mesh-explorer (as a private version)?

**Reply by pazner** [reply]
Something like the following diff may help you. With this patch, use `V` in mesh-explorer to write to VisIt format
```diff
diff --git a/miniapps/meshing/mesh-explorer.cpp b/miniapps/meshing/mesh-explorer.cpp
index c0ca387a4..872f1abe6 100644
--- a/miniapps/meshing/mesh-explorer.cpp
+++ b/miniapps/meshing/mesh-explorer.cpp
@@ -383,7 +383,7 @@ int main (int argc, char *argv[])
            "p) Generate a partitioning\n"
            "o) Reorder elements\n"
            "S) Save in MFEM format\n"
-           "V) Save in VTK format (only linear and quadratic meshes)\n"
+           "V) Save in VisIt format\n"
            "q) Quit\n"
 #ifdef MFEM_USE_ZLIB
            "Z) Save in MFEM format with compression\n"
@@ -1183,11 +1183,11 @@ int main (int argc, char *argv[])
 
       if (mk == 'V')
       {
-         const char omesh_file[] = "mesh-explorer.vtk";
-         ofstream omesh(omesh_file);
-         omesh.precision(14);
-         mesh->PrintVTK(omesh);
-         cout << "New VTK mesh file: " << omesh_file << endl;
+         const char omesh_file[] = "mesh-explorer";
+         VisItDataCollection dc(omesh_file, mesh);
+         dc.SetPrecision(14);
+         dc.Save();
+         cout << "New VisIt mesh file: " << omesh_file << "_000000.mfem_root"
+              << endl;
       }
```

**Reply by wnissen** [reply]
Wow, let me try that! Thanks for giving it to me on a silver platter.

**Reply by v-dobrev** [reply]
We should add this menu options to `mesh-explorer` -- both VisIt and Paraview exports will be useful.

**Reply by wnissen** [reply]
Yes, that would be handy. I was able to get it read into VisIt!
![image](https://user-images.githubusercontent.com/1278310/211112371-61c7e17c-7653-4724-94f5-c3ba4822b87a.png)
I'm not sure there is a common palette between GLvis and VisIt, but this is close. I thought I knew what it meant by `h) View element sizes, h` but maybe h isn't the volume? Seems highly correlated, at least.
